src/main/java/org/fcrepo/camel/processor/ProcessorUtils.java

```diff
@@ -40,55 +42,24 @@ private ProcessorUtils() {
     }
 
     private static String trimTrailingSlash(final String path) {
-        String trimmed = path;
-        while (trimmed.endsWith("/")) {
-            trimmed = trimmed.substring(0, trimmed.length() - 1);
-        }
-        return trimmed;
-    }
-
-    /**
-     * Extract a language string suitable for Jena from a MimeType value
-     * @param contentType a MIMEType string
-     * @return a Jena-compatible language string
-     */
-    public static String langFromMimeType(final String contentType) {
-        if (contentType.equals("application/n-triples")) {
-            return "N-TRIPLE";
-        } else if (contentType.equals("text/turtle")) {
-            return "TURTLE";
-        } else if (contentType.equals("application/rdf+xml")) {
-            return "RDF/XML";
-        } else if (contentType.equals("text/rdf+n3") || contentType.equals("text/n3")) {
-            return "N3";
-        } else {
-            return null;
+        if (path.endsWith("/")) {
+            return trimTrailingSlash(path.substring(0, path.length() - 1));
         }
+        return path;
     }
 
     /**
      * Extract the subject URI from the incoming message headers.
      * @param in the incoming Message
      * @return the subject URI
-     * @throws IOException when no baseURL header is present
      */
-    public static String getSubjectUri(final Message in) throws IOException {
-        final StringBuilder base = new StringBuilder("");
-
-        if (in.getHeader(FcrepoHeaders.FCREPO_BASE_URL) != null) {
-            base.append(trimTrailingSlash(in.getHeader(FcrepoHeaders.FCREPO_BASE_URL, String.class)));
-        } else if (in.getHeader(JmsHeaders.BASE_URL) != null) {
-            base.append(trimTrailingSlash(in.getHeader(JmsHeaders.BASE_URL, String.class)));
-        } else {
-            throw new IOException("No baseURL header available!");
-        }
-
-        if (in.getHeader(FcrepoHeaders.FCREPO_IDENTIFIER) != null) {
-           base.append(in.getHeader(FcrepoHeaders.FCREPO_IDENTIFIER, String.class));
-        } else if (in.getHeader(JmsHeaders.IDENTIFIER) != null) {
-           base.append(in.getHeader(JmsHeaders.IDENTIFIER, String.class));
+    public static Optional<String> getSubjectUri(final Message in) {
```

Comment:
Yes, there is. The subject URI should come from the camel-specific headers "CamelFcrepoBaseUrl", etc, and not from the JMS-specific headers (i.e. I'd like to see non-JMS based message brokers, and I don't want fcrepo-camel tied to a particular broker). That means that those message headers need to be added -- either from processing the incoming JSON message body (`unmarshal().json(Jackson).setHeader(...)`) or using an implementation-specific header such as `setHeader(...).header("org.fcrepo.jms.identifier")`. The current code will _act_ on those JMS-based headers, and that's exactly what I want to remove.